qiskit_ibm_runtime/base_runtime_job.py

```diff
@@ -255,7 +258,7 @@ def _set_error_message(self, job_response: Dict) -> None:
         Args:
             job_response: Job response from runtime API.
         """
-        if self.errored():
+        if self._status == self.ERROR:
```

Comment:
Same as above, I'm not sure why this change is required.

Reply:
Same reasoning as above, it would result in an infinite loop